query_optimizer/resolver/Resolver.cpp

```diff
@@ -2471,6 +2679,154 @@ E::ScalarPtr Resolver::resolveFunctionCall(
   return E::ToRef(aggregate_alias);
 }
 
+E::ScalarPtr Resolver::resolveWindowAggregateFunction(
```

Comment:
I see this new method shares too many common code with `Resolver::resolveFunctionCall`, so I would suggest to combine them.